doc/usecases/elektra_web/UC_key_validation.md

```diff
@@ -0,0 +1,25 @@
+# Use Case: Key validation
+
+## Summary
+
+Title: Key validation
+Scope: Configuration
+Level: User Goal
+Actors: User (usually a sysadmin)
+Brief: User gets feedback about a key's contents.
+
+## Scenarios
+
+Precondition: Modifying keys
+Main success scenario: User edits a key's contents in a field that is
+  specifically formatted according to its metadata (e.g. number/date fields vs
+  string fields with regex validation). If the entered data passes the
+  validation, it is sent to the backend and saved to the key database.
+Alternative scenario: User enters data that fails the validation, the input
+  field is marked as invalid (e.g. with a red color and some information text)
+  and the data is NOT sent to the backend.
+Error scenario: Technical problems while persisting to the key database. The
```

Comment:
Maybe you are trying to say too much with the use cases? I only noticed several times that it is very easy to find inconsistencies and points where clarifications are needed.

For example, now that you say it: undo is supported after every other *use case* (e.g. finding keys)? (Rather it only applies to actions that changed the state of the system.)

In the same way the validation only applies when the user edits something in the tree view. It shouldn't be too hard to say that explicitly.

Basically I am okay with every description that:

1. clearly describes what the user can achieve with your proposed system
2. are internally consistent